pkg/client/patch.go

```diff
@@ -211,3 +212,34 @@ func (p applyPatch) Data(obj Object) ([]byte, error) {
 	// client-go does, more-or-less).
 	return json.Marshal(obj)
 }
+
+//applyFromPatch uses server-side-apply with a ApplyConfiguration object.
+type applyFromPatch struct {
+	patch interface{}
+}
+
+// Type implements Patch.
+func (p applyFromPatch) Type() types.PatchType {
+	return types.ApplyPatchType
+}
+
+// Sets the name and namespace for the returned object if an ApplyConfiguration was supplied
```

Comment:
very minor nit, still follow godoc conventions here? E.g. "setNameNamespace sets the ..."